core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -66,6 +77,29 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     }
     sc = new SparkContext(conf)
     taskScheduler = new TaskSchedulerImpl(sc)
+    setupHelper()
+  }
+
+  def setupSchedulerWithMockTsm(): TaskSchedulerImpl = {
```

Comment:
Also, where is the spied TSM actually used? (can you add a comment about this) From a quick skim I could only find the blacklist methods being checked.